tests/ops/op_math/test_dot.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the dot function
+"""
+import pytest
+
+import pennylane as qml
+from pennylane.ops import SProd, Sum, dot
+from pennylane.pauli.pauli_arithmetic import PauliSentence
+
+coeffs_and_ops_list = [
+    ([1.0], [qml.PauliX(0)]),
+    ([1.1, 2.2], [qml.PauliX(0), qml.PauliY(1)]),
+    ([1.1, 2.2, 3.3], [qml.PauliX(0), qml.PauliY(1), qml.PauliZ(2)]),
+]
+
+
+class TestDot:
+    """Unittests for the dot function."""
+
+    def test_dot_returns_pauli_sentence(self):
+        """Test that the dot function returns a PauliSentence class when ``pauli=True``."""
+        coeffs = [1.0, 2.0, 3.0]
```

Comment:
don't forget to also test this for numpy, tf, torch, jax tensors :)

Reply:
Added tests! 👍